cmd/mayactl/app/command/volume_info.go

```diff
@@ -0,0 +1,105 @@
+package command
+
+import (
+	"errors"
+	"fmt"
+	client "github.com/openebs/maya/pkg/client/jiva"
+	"github.com/openebs/maya/pkg/util"
+	"github.com/openebs/maya/types/v1"
+	"github.com/spf13/cobra"
+	"os"
+	"strings"
+	"text/tabwriter"
+)
+
+var (
+	volumeInfoCommandHelpText = `
+	    Usage: mayactl volume info --volname <vol>
+
+            This command queries the info of the volume.
+        `
+)
+
+// CmdVolumeAccessOptions is used to store the value of flags used in the cli
+type CmdVolumeInfoOptions struct {
+	volName string
+}
+
+// NewCmdVolumeAccessMode shows info of OpenEBS Volume
+func NewCmdVolumeInfo() *cobra.Command {
+	options := CmdVolumeInfoOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "info",
+		Short:   "Displays the info of Volume",
+		Long:    volumeInfoCommandHelpText,
+		Example: `mayactl volume info --volname vol`,
+		Run: func(cmd *cobra.Command, args []string) {
+			util.CheckErr(options.Validate(cmd), util.Fatal)
+			util.CheckErr(options.RunVolumeInfo(cmd), util.Fatal)
+		},
+	}
+
+	cmd.Flags().StringVarP(&options.volName, "volname", "n", options.volName,
+		"unique volume name.")
+	return cmd
+}
+
+func (c *CmdVolumeInfoOptions) Validate(cmd *cobra.Command) error {
+	if c.volName == "" {
+		return errors.New("--volname is missing. Please specify a volume name created")
+	}
+	return nil
+}
+
+// TODO : Add more volume information
+
+// RunVolumeAccessMode runs info command and display the outputs in standard I/O.
+// Currently It displays volume access mode only.
+func (c *CmdVolumeInfoOptions) RunVolumeInfo(cmd *cobra.Command) error {
+	annotation := &Annotations{}
+	// GetVolumeAnnotation is called to get the volume controller's info such as
+	// controller's IP, status, iqn, replica IPs etc.
+	err := annotation.GetVolAnnotations(c.volName)
+	if err != nil {
+		fmt.Println("Can't get annotation, found error ", err)
+	}
+
+	if err != nil || annotation == nil {
+		fmt.Println(err)
+		return nil
+	}
+
+	if annotation.ControllerStatus != "Running" {
+		fmt.Println("Volume not reachable")
+		return nil
+	}
+	// Initiallize an instance of ReplicaCollection, json response recieved from the
+	// controllerIP:9501/v1/replicas is to be parsed into this structure via GetVolumeAccessMode.
+	// GetVolumeAccessMode only gives the details related to the access modes.
+
+	collection := client.ReplicaCollection{}
+	err = client.GetVolumeAccessMode(annotation.ClusterIP+":9501", &collection)
```

Comment:
Use variables for numeric ports unless the ports 9501 and 9502 are going to be constant.

Reply:
@avishnu these are the constants.